django/core/checks/settings_deprecation.py

```diff
@@ -0,0 +1,87 @@
+from django.conf import settings
+from django.core import checks
+
+
+def deprecated_settings_variables(*args, **kwargs):
+    """
+    This check warns users who still use deprecated settings variables.
+    """
+
+    deprecated_settings_list = (
+        # todo: add older settings
+        # Django 1.4
+        "TRANSACTIONS_MANAGED",
+        # Django 1.5
+        "SITE_ID",
+        "AUTH_PROFILE_MODULE",
+        "CSRF_COOKIE_PATH",
+        # Django 1.6
+        "DATABASE_ENGINE",
+        # Django 1.7
+        "SOUTH_DATABASE_ADAPTER",
+        "SOUTH_DATABASE_ADAPTERS",
+        "SOUTH_AUTO_FREEZE_APP",
+        "SOUTH_TESTS_MIGRATE",
+        "SOUTH_LOGGING_ON",
+        "SOUTH_LOGGING_FILE",
+        "SOUTH_MIGRATION_MODULES",
+        "SOUTH_USE_PYC",
+        # Django 1.8
+        "SEND_BROKEN_LINK_EMAILS",
+        "CACHE_MIDDLEWARE_ANONYMOUS_ONLY",
+        # Django 1.9
+        "FILE_UPLOAD_HANDLERS",
+        # Django 1.10
+        "ALLOWED_INCLUDE_ROOTS",
+        "LOGOUT_URL"
+        "SECURE_PROXY_SSL_HEADER",
+        "TEMPLATE_CONTEXT_PROCESSORS",
+        "TEMPLATE_DEBUG",
+        "TEMPLATE_DIRS",
+        "TEMPLATE_LOADERS",
+        "TEMPLATE_STRING_IF_INVALID",
+        # Django 2.1
+        "USE_ETAGS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 3.0
+        "DEFAULT_CONTENT_TYPE",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        # Django 3.1
+        "DEFAULT_FILE_STORAGE",
+        "FILE_CHARSET",
+        # Django 4.0
+        "DEFAULT_HASHING_ALGORITHM",
+        "PASSWORD_RESET_TIMEOUT_DAYS",
+        "SECURE_BROWSER_XSS_FILTER",
+        # Django 4.1
+        "DEFAULT_STORAGE_CLASS",
+        # Django 4.2
+        "MIDDLEWARE_CLASSES",
+        "FILE_UPLOAD_HANDLERS",
+        "ADMINS",
+        "MANAGERS",
+        # Django 5.0
+        "USE_L10N",
+        "USE_DEPRECATED_PYTZ",
+        "CSRF_COOKIE_MASKED",
+        "DATABASE_OPTIONS",
+        # todo: DATABASES->name->TEST->SERIALIZE not yet covered
+        # Django 5.1
+        "DEFAULT_FILE_STORAGE",
+        "STATICFILES_STORAGE",
+    )
```

Comment:
```suggestion
    deprecated_settings_list = {
        # todo: add older settings
        # Django 1.4
        "TRANSACTIONS_MANAGED",
        # Django 1.5
        "SITE_ID",
        "AUTH_PROFILE_MODULE",
        "CSRF_COOKIE_PATH",
        # Django 1.6
        "DATABASE_ENGINE",
        # Django 1.7
        "SOUTH_DATABASE_ADAPTER",
        "SOUTH_DATABASE_ADAPTERS",
        "SOUTH_AUTO_FREEZE_APP",
        "SOUTH_TESTS_MIGRATE",
        "SOUTH_LOGGING_ON",
        "SOUTH_LOGGING_FILE",
        "SOUTH_MIGRATION_MODULES",
        "SOUTH_USE_PYC",
        # Django 1.8
        "SEND_BROKEN_LINK_EMAILS",
        "CACHE_MIDDLEWARE_ANONYMOUS_ONLY",
        # Django 1.9
        "FILE_UPLOAD_HANDLERS",
        # Django 1.10
        "ALLOWED_INCLUDE_ROOTS",
        "LOGOUT_URL"
        "SECURE_PROXY_SSL_HEADER",
        "TEMPLATE_CONTEXT_PROCESSORS",
        "TEMPLATE_DEBUG",
        "TEMPLATE_DIRS",
        "TEMPLATE_LOADERS",
        "TEMPLATE_STRING_IF_INVALID",
        # Django 2.1
        "USE_ETAGS",
        "SECURE_BROWSER_XSS_FILTER",
        # Django 3.0
        "DEFAULT_CONTENT_TYPE",
        "PASSWORD_RESET_TIMEOUT_DAYS",
        # Django 3.1
        "DEFAULT_FILE_STORAGE",
        "FILE_CHARSET",
        # Django 4.0
        "DEFAULT_HASHING_ALGORITHM",
        "PASSWORD_RESET_TIMEOUT_DAYS",
        "SECURE_BROWSER_XSS_FILTER",
        # Django 4.1
        "DEFAULT_STORAGE_CLASS",
        # Django 4.2
        "MIDDLEWARE_CLASSES",
        "FILE_UPLOAD_HANDLERS",
        "ADMINS",
        "MANAGERS",
        # Django 5.0
        "USE_L10N",
        "USE_DEPRECATED_PYTZ",
        "CSRF_COOKIE_MASKED",
        "DATABASE_OPTIONS",
        # todo: DATABASES->name->TEST->SERIALIZE not yet covered
        # Django 5.1
        # https://docs.djangoproject.com/en/stable/releases/5.1/#features-removed-in-5-1
        "DEFAULT_FILE_STORAGE",
        "STATICFILES_STORAGE",
    }
```
Very useful PR :)

It will be useful to add link to the paragraph of feature removed in release notes (see the example for Dajngo 5.1) 

Using `set` instead of `tuple` will be 10x faster:
```pycon
>>> type(deprecated_settings_list)
tuple
>>> %timeit bool("ADMINS" in deprecated_settings_list)
170 ns ± 2.24 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
>>> deprecated_settings_set = set(deprecated_settings_list)
>>> type(deprecated_settings_set)
set
>>> %timeit bool("ADMINS" in deprecated_settings_set)
18.1 ns ± 0.229 ns per loop (mean ± std. dev. of 7 runs, 100,000,000 loops each)
```

Reply:
Awesome! Applied your suggestion. Have to keep that in mind 👌